src/masternodes/validation.cpp

```diff
@@ -2382,67 +2382,69 @@ static void RevertFailedTransferDomainTxs(const std::vector<std::string> &failed
-            assert(id);
-            const auto node = cache.GetMasternode(*id);
-            assert(node);
-
-            auto height = node->creationHeight;
-            auto mnID = *id;
-            if (!node->collateralTx.IsNull()) {
-                const auto idHeight = cache.GetNewCollateral(node->collateralTx);
-                assert(idHeight);
-                height = idHeight->blockHeight - GetMnResignDelay(std::numeric_limits<int>::max());
-                mnID = node->collateralTx;
-            }
-
-            const auto blockindex = ::ChainActive()[height];
-            assert(blockindex);
-
-            CTransactionRef tx;
-            uint256 hash_block;
-            assert(GetTransaction(mnID, tx, Params().GetConsensus(), hash_block, blockindex));
-            assert(tx->vout.size() >= 2);
-
-            CTxDestination dest;
-            assert(ExtractDestination(tx->vout[1].scriptPubKey, dest));
-            assert(dest.index() == PKHashType || dest.index() == WitV0KeyHashType);
-
-            const auto keyID = dest.index() == PKHashType ? CKeyID(std::get<PKHash>(dest)) : CKeyID(std::get<WitnessV0KeyHash>(dest));
-            std::copy(keyID.begin(), keyID.end(), beneficiary.begin());
-            minerAddress = GetScriptForDestination(dest);
-        } else {
-            std::copy(minter.begin(), minter.end(), beneficiary.begin());
-            const auto dest = PKHash(minter);
-            minerAddress = GetScriptForDestination(dest);
-        }
-
-        CrossBoundaryResult result;
-        const auto blockResult = evm_try_finalize(result, evmContext, false, block.nBits, beneficiary, block.GetBlockTime());
-        auto evmBlockHash = std::vector<uint8_t>(blockResult.block_hash.begin(), blockResult.block_hash.end());
-        std::reverse(evmBlockHash.begin(), evmBlockHash.end());
-
-        cache.SetVMDomainBlockEdge(VMDomainEdge::DVMToEVM, block.GetHash(), uint256(evmBlockHash));
-        cache.SetVMDomainBlockEdge(VMDomainEdge::EVMToDVM, uint256(evmBlockHash), block.GetHash());
-
-        if (!blockResult.failed_transactions.empty()) {
-            std::vector<std::string> failedTransactions;
-            for (const auto& rust_string : blockResult.failed_transactions) {
-                failedTransactions.emplace_back(rust_string.data(), rust_string.length());
-            }
+    CrossBoundaryResult result;
+    const auto blockResult = evm_try_finalize(result, evmContext, false, block.nBits, beneficiary, block.GetBlockTime());
+    if (!result.ok) {
+        LogPrintf("ERROR: EVM try finalize failed: %s\n", result.reason.c_str());
+    }
+    auto evmBlockHashData = std::vector<uint8_t>(blockResult.block_hash.rbegin(), blockResult.block_hash.rend());
+    auto evmBlockHash = uint256(evmBlockHashData);
+    
+    cache.SetVMDomainBlockEdge(VMDomainEdge::DVMToEVM, block.GetHash(), evmBlockHash);
+    cache.SetVMDomainBlockEdge(VMDomainEdge::EVMToDVM, evmBlockHash, block.GetHash());
```

Comment:
Just marking out this for the diff as this area has been changed. Please resolve comment on review.